ast.ts

```diff
@@ -42,6 +43,7 @@ export type Expr<A> =
   | {  a?: A, tag: "index", obj: Expr<A>, index: Expr<A> }
   | {  a?: A, tag: "method-call", obj: Expr<A>, method: string, arguments: Array<Expr<A>> }
   | {  a?: A, tag: "construct", name: string }
+  | {  a?: A, tag: "bracket", values: Array<Expr<A>>}
```

Comment:
What's this expression? If you add something to the AST, you should describe somewhere in your write-up what it's supposed to mean.

Reply:
I changed the name back to "set", as written in our design.md. We will update the pull request later today.